core/primitives/src/trie_key.rs

```diff
@@ -250,6 +269,14 @@ impl TrieKey {
                 buf.push(ACCOUNT_DATA_SEPARATOR);
                 buf.extend(data_id.as_ref());
             }
+            TrieKey::BufferedReceiptIndices => buf.push(col::BUFFERED_RECEIPT_INDICES),
+            TrieKey::BufferedReceipt { index, receiving_shard } => {
+                buf.push(col::BUFFERED_RECEIPT);
+                // Use  u16 for shard id to reduce depth in trie.
+                assert!(*receiving_shard <= u16::MAX as u64, "Shard ID too big.");
```

Comment:
totally optional suggestion:
You can adding a unit test to make sure this gets caught. The test can get the shard layout for the PROTOCOL_VERSION and check that max shard id meets this condition. Better to check the max rather than len because we may change the numbering.

Reply:
This would be a great test, yeah. But it seems there is no trivial way to get a shard layout for the latest protocol version right now, so let me add that in a separate PR.

Reply:
https://github.com/near/nearcore/pull/11186